source/extensions/filters/udp/udp_proxy/router/router.h

```diff
@@ -26,11 +26,11 @@ class Router {
    * @return the cluster name or empty string if there is not matching route for the data.
    */
   virtual const std::string
-  route(Network::Address::InstanceConstSharedPtr destination_address,
-        Network::Address::InstanceConstSharedPtr source_address) const PURE;
+  route(const Network::Address::InstanceConstSharedPtr destination_address,
```

Comment:
Sorry I meant `const Network::Address::Instance&`

Reply:
https://github.com/envoyproxy/envoy/blob/eede13d30efceb3e8c9fb3bb3e39a527495a86f9/source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc#L50

In the caller method, we will get a `InstanceConstSharedPtr`. If we change the `Router::route`'s signature to `const Network::Address::Instance&`, we have to change the code in the caller to

```cpp
const std::string& route = config_->route(*data.addresses_.local_, *data.addresses_.peer_); 
```

Do you think it is acceptable?

Reply:
Fixed, I have updated all methods in the caller chain.